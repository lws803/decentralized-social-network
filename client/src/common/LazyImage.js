import React from "react";

import { LazyLoadImage } from "react-lazy-load-image-component";

import Placeholder from "../res/placeholder_round.png";

export default function LazyImage(props) {
  return (
    <LazyLoadImage
      {...props}
      effect="blur"
      placeholderSrc={Placeholder}
    />
  );
}
