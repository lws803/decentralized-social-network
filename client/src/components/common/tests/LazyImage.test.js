import React from "react";
import Enzyme, { shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import { createSerializer } from "enzyme-to-json";

import LazyImage from "../LazyImage";
import ProfilePhoto from "../../../res/7874219.jpeg";

Enzyme.configure({ adapter: new Adapter() });
expect.addSnapshotSerializer(createSerializer({ mode: "deep" }));

describe("Lazy Image component", () => {
  it("renders correctly", () => {
    const wrapper = shallow(<LazyImage src={ProfilePhoto}/>);
    expect(wrapper).toMatchSnapshot();
  })
})
