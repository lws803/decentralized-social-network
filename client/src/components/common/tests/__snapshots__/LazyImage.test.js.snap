// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Lazy Image component renders correctly 1`] = `
<t
  afterLoad={[Function]}
  beforeLoad={[Function]}
  delayMethod="throttle"
  delayTime={300}
  effect="opacity"
  placeholderSrc={null}
  src="7874219.jpeg"
  threshold={100}
  useIntersectionObserver={true}
  visibleByDefault={false}
  wrapperClassName=""
/>
`;
